libs/core/synchronization/include/hpx/synchronization/mutex.hpp

```diff
@@ -32,75 +33,398 @@ namespace hpx { namespace threads {
 
 namespace hpx {
 
-    ///////////////////////////////////////////////////////////////////////////
+    ///
+    /// \brief \a mutex class is a synchronization primitive that can be used
+    ///        to protect shared data from being simultaneously accessed by
+    ///        multiple threads.
+    ///        \a mutex offers exclusive, non-recursive ownership semantics:
+    ///
+    ///        - A calling thread owns a mutex from the time that it
+    ///          successfully calls either \a lock or \a try_lock until it
+    ///          calls \a unlock.
+    ///        - When a thread owns a \a mutex, all other threads will
+    ///          block (for calls to \a lock) or receive a \a false return
+    ///          value (for \a try_lock) if they attempt to claim ownership
+    ///          of the \a mutex.
+    ///        - A calling thread must not own the \a mutex prior to
+    ///          calling \a lock or \a try_lock.
+    ///
+    ///        The behavior of a program is undefined if a \a mutex is
+    ///        destroyed while still owned by any threads, or a thread
+    ///        terminates while owning a \a mutex. The mutex class satisfies
+    ///        all requirements of \a Mutex and \a StandardLayoutType.
+    ///
+    ///        \a hpx::mutex is neither copyable nor movable.
+    ///
     class mutex
     {
     public:
+        /// \brief \a hpx::mutex is neither copyable nor movable
         HPX_NON_COPYABLE(mutex);
 
     protected:
+        /// \cond NOPROTECTED
         typedef hpx::spinlock mutex_type;
+        /// \endcond NOPROTECTED
 
     public:
+        ///
+        /// \brief Constructs the \a mutex. The \a mutex is in unlocked state
+        ///        after the constructor completes.
+        ///
+        ///       FIXME(bhumit->hkaiser): true for hpx::mutex?
+        /// \note Because the default constructor is \a constexpr, static
```

Comment:
Is this true for hpx::mutex as well?

Reply:
Good catch, we should apply the corresponding changes. Could you please create a ticket for us not to forget? In the end, all we need is to add a constexpr default constructor and leave the one that takes the (currently optional) string in place without a default value.

Reply:
Will do.